test/state.test.ts

```diff
@@ -1,200 +1,289 @@
-                "1101952250596507940009480237111758916141356508680126512170143672832985442917"
-            ]
-        }
-
-        const expectedErrorText = "oldState argument should be equal to the latest identity state in smart contract when isOldStateGenesis == 0";
-        let isException = false;
-        try {
-            await state.transitState(params.id, params.oldState, params.newState, params.isOldStateGenesis, params.a, params.b, params.c);
-        } catch (e: any) {
-            isException = true;
-            expect(e.message).contains(expectedErrorText);
-        }
-        expect(isException).to.equal(true)
-
-        const res = await state.getState(params.id);
-        expect(res.toString()).to.not.be.equal(bigInt(params.newState).toString());
-    });
-
-    it("Negative: state publishing with isOldStateGenesis = 0 for initial state publishing", async () => {
-        const params = {
-            "id": "329949150130214723420589610911161895495647789006649785264738141299135414272",
-            "oldState": "8061408109549794622894897529509400209321866093562736009325703847306244896707",
-            "newState": "5451025638486093373823263243878919389573792510506430020873967410859218112302",
-            "isOldStateGenesis": false,
-            "a": [
-                "9123530042567533646828922425772487350426456162482164386394882200839785942640",
-                "9415307921947273965252020837563837769736935217738655289285974217793588520580",
-            ],
-            "b": [
-                [
-                    "20935778885211506570223376266828471506430371932508832162652558992203635947966",
-                    "5508818874264996604372631831086863512981873969182281232851779203198165199858"
-                ],
-                [
-                    "7582649530445089244527470093316718411976974567026715956160946701678788144099",
-                    "15328003255722578273426427675683596906331019433013116900725008315768237010126"
-                ]
-            ],
-            "c": [
-                "11960369619177612085202166474468359141481452033953473766663574668254793582474",
-                "2499582274802148691487974614055461818489209711650283363834328284825343196632",
-            ],
-        };
-
-
-        const expectedErrorText = "there should be at least one state for identity in smart contract when isOldStateGenesis == 0";
-        let isException = false;
-        try {
-            await state.transitState(params.id, params.oldState, params.newState, params.isOldStateGenesis, params.a, params.b, params.c);
-        } catch (e: any) {
-            isException = true;
-            expect(e.message).contains(expectedErrorText);
-        }
-        expect(isException).to.equal(true)
-
-        const res = await state.getState(params.id);
-        expect(res.toString()).to.be.equal("0");
-    });
+  let verifier;
+  let state;
+
+  before(async () => {
+    const Verifier = await ethers.getContractFactory("Verifier");
```

Comment:
This part of the code becomes too bulky. Why not to put it into the utils or reuse existing `deployContracts` from utils

Reply:
fixed